base/include/BoundBuffer.h

```diff
@@ -38,6 +38,25 @@ class bounded_buffer
 		}
 	}
 
+	void push_back(typename boost::call_traits<value_type>::param_type item)
+	{ // `param_type` represents the "best" way to pass a parameter of type `value_type` to a method.
+
+		boost::mutex::scoped_lock lock(m_mutex);
+		m_not_full.wait(lock, boost::bind(&bounded_buffer<value_type>::is_ready_to_accept, this));
+		if (is_not_full() && m_accept)
```

Comment:
If a queue is full & the command is not enqueued, it is not a priority command. 
We should do this for prioritiy commands:
- if size of queue < 2 * queueSize 
- then enqueue the command frame & let the queueSize increase

Reply:
2 * queueSize check is just for safety

Reply:
Added the changes